uaclient/util.py

```diff
@@ -218,6 +219,37 @@ def disable_log_to_console():
         console_handler.setLevel(old_level)
 
 
+def retry(exception, retry_sleeps):
+    """Decorator to retry on exception for retry_sleeps.
+
+     @param retry_sleeps: List of sleep lengths to apply between
+        retries. Specifying a list of [0.5, 1] tells subp to retry twice
+        on failure; sleeping half a second before the first retry and 1 second
+        before the second retry.
+     @param exception: The exception class to catch and retry for the provided
+        retry_sleeps. Any other exception types will immediately be raised.
```

Comment:
A nit, but they won't be caught at all, which is a little different to being raised.